### PR TITLE
17.3

### LINKED ISSUE #1574: Invoice Ninja - Set "API_SECRET" on firstboot
[Lee reported in the forums](https://www.turnkeylinux.org/comment/49283#comment-49283) that the Invoice Ninja `API_SECRET` key needs to be manually set before you can use the Invoice Ninja API (e.g. via the Invoice Ninja Android app).

It's not a security issue because ti won't allow connections until it's set. However, it's not very TurnKey to have to manually do that!

After having a closer look, it _should_ be being set by the [`php $WEBROOT/artisan key:generate --force`](https://github.com/turnkeylinux-apps/invoice-ninja/blob/master/overlay/usr/lib/inithooks/firstboot.d/20regen-ninja-invoice-secrets#L18) line, but it seems that's not working as it should?! So it seems we may need to manually/explicitly set that.

### LINKED ISSUE #1794: Invoice Ninja - Can't preview & print PDFs
Using the 17.2 lxc template in proxmox (which can be downloaded directly via their web UI), I can't preview or print PDFs.
The original error states, that `libnss3` is missing. Installing that shows more dependencies missing. 
The following thread helped with installing all the necessary dependencies: https://forum.invoiceninja.com/t/new-install-cant-print-dl-invoices/12193/6. After that I was able to view and print PDFs. 
The person there said, that they made a PR, but I couldn't find it.

### PR DESCRIPTION
**Pull request by JedMeister**
- Update Invoice Ninja
- Pre-install SnapPDF dependencies (so PDF generation actually works) - closes https://github.com/turnkeylinux/tracker/issues/1794
- Set API_SECRET on firstboot - closes https://github.com/turnkeylinux/tracker/issues/1574
- Fix cron job (untracked issue) - also requires: https://github.com/turnkeylinux/common/pull/241

### CODE REVIEW

File: conf.d/main [hunk 1]
```diff
@@ -83,14 +83,23 @@ _token=$TOKEN\
 &terms_of_service=on\
 &privacy_policy=on"
 
-echo '  * * * * * www-data php /var/www/invoiceninja/artisan schedule:run >> /dev/null 2>&1' >> /etc/cron.d/invoice-ninja
+echo '  * * * * * root /usr/local/bin/turnkey-artisan schedule:run >> /dev/null 2>&1' >> /etc/cron.d/invoice-ninja
```
Comment (OnGle) [inline conf.d/main]: If this change to using root is intentional, could you add a comment explaining why?

**Review by OnGle — CHANGES_REQUESTED** [review]
Small tweak suggested, functionally happy with this PR.

File: conf.d/main (hunk 1 above)
Comment (JedMeister) [reply conf.d/main]: Good call. Will do.

**Review by Christie018 — APPROVED** [review]
Code changes look good to me and are documented in the appliance changelog,.

**Comment by JedMeister** [comment]
Thanks to both of you.

**Pull request merged**